core/src/proxy/block_engine_stage.rs

```diff
@@ -300,10 +369,36 @@ impl BlockEngineStage {
                     block_engine_stats = BlockEngineStageStats::default();
                 }
                 _ = maintenance_tick.tick() => {
-                    let block_builder_info = client.get_block_builder_fee_info(BlockBuilderFeeInfoRequest{}).await?.into_inner();
+
+                    if cluster_info.id() != keypair.pubkey() {
```

Comment:
imo we should generate a signal to catch this faster. need to provide a signal mechanism from set_identity

Reply:
less hacky would be doing the check once per second

Reply:
Checking once per second now.  Will explore other signal mechanisms in followup pr for dynamic back end config changes.

Reply:
Checking once per second now.  Will explore other signal mechanisms in followup pr for dynamic back end config changes.